### ISSUE TITLE
`pkgConf` has no effect after the first usage

### ISSUE THREAD

**Issue by fasttime** [yargs/yargs#2191]
**yargs v17.5.1**

Currently, `pkgConf` only loads package.json options when it is called for the first time. All other usages are silently ignored. This makes it impossible to reuse the same instance of Yargs when the `cwd` or the `key` value changes.
```js
const Yargs = require('yargs/yargs')([]);
const opts1 = Yargs.pkgConf('test', 'a').parse();
const opts2 = Yargs.pkgConf('test', 'b').parse();
const opts3 = Yargs.pkgConf('test2', 'a').parse();
console.log({ opts1, opts2, opts3 }); // opts1, opts2 and opts3 are the same(!)
```

In the example above, `opts2` and `opts3` always end up getting the same values as `opts1`, because only the first usage of `pkgConf` has an effect.

It would be helpful if those repeated usages would produce an error telling the user that they are doing something wrong, or even better, if using `pkgConf` multiple times was allowed.

**Comment by shadowspawn** [comment]
You should make a new yargs instance for every parse. 

```js
const Yargs = require('yargs/yargs');
const opts1 = Yargs([]).pkgConf('test', 'a').parse();
const opts2 = Yargs([]).pkgConf('test', 'b').parse();
```

**Comment by fasttime** [comment]
> You should make a new yargs instance for every parse.

Yeah, I thought it would be helpful if yargs would throw an error with that message. Although my assumption was that `pkgConf` would be more of a problem than `parse` when called multiple times.

**Comment by shadowspawn** [comment]
I see what you mean, but I expect there are multiple routines that don't work when reuse same instance for another test. (There used to be a `.reset()` method but it is deprecated and gone.) The current recommendation is start again with a fresh instance for each test.

**Comment by shadowspawn** [comment]
A clarification: I am not sure if `.pkgConf()` can be called multiple times, but I am sure you should not reuse yargs instance after calling `.parse()`. So only commenting on part of your scenario and observations.

**Comment by fasttime** [comment]
Thanks for the clarification. If so, I wonder if it would be a breaking change to throw an error - or at least to print a warning message - when `.parse()` is called multiple times on an instance. That would have saved me from opening this issue.

**Comment by shadowspawn** [comment]
It would definitely be a breaking change in terms of requiring a new major version.

But interesting suggestion which gets to the heart of the issue. (And I hadn't though of it that way.)

**Comment by shadowspawn** [comment]
Full disclosure, I came across a PR tonight that was making one particular situation allow calling parse twice! #1369 

(I stick by my advice to make a new instance for each test, but less sure what is and isn't expected to work...)

**Comment by shadowspawn** [comment]
I misunderstood that Yargs does mostly make `.parse()` a non-breaking call. I am going to change my earlier "should make a new yargs instance for every parse" to "could...", hide some of my intermediate answers, and take another look at some point!
